addons/binding/org.openhab.binding.ikeatradfri/src/main/java/org/openhab/binding/ikeatradfri/handler/IkeaTradfriBulbHandler.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.ikeatradfri.handler;
+
+import static org.openhab.binding.ikeatradfri.IkeaTradfriBindingConstants.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.ikeatradfri.internal.IkeaTradfriObserveListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link IkeaTradfriBulbHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniel Sundberg - Initial contribution
+ */
+public class IkeaTradfriBulbHandler extends BaseThingHandler implements IkeaTradfriObserveListener {
+
+    private class LightProperties {
+        JsonObject root;
+        JsonObject settings;
+        JsonArray array;
+
+        public LightProperties() {
+            root = new JsonObject();
+            settings = new JsonObject();
+            array = new JsonArray();
+            array.add(settings);
+            root.add(TRADFRI_LIGHT, array);
+        }
+
+        public LightProperties(JsonElement json) {
+            try {
+                root = json.getAsJsonObject();
+                array = root.getAsJsonArray(TRADFRI_LIGHT);
+                settings = array.get(0).getAsJsonObject();
+            } catch (JsonSyntaxException e) {
+                logger.error("JSON error: {}", e.getMessage());
+            }
+        }
+
+        LightProperties setBrightness(PercentType b) {
+            settings.add(TRADFRI_DIMMER, new JsonPrimitive(Math.round(b.floatValue() / 100.0f * 254)));
+            return this;
+        }
+
+        PercentType getBrightness() {
+            int b = settings.get(TRADFRI_DIMMER).getAsInt();
+            if (b == 1) {
```

Comment:
Depends on preference I guess.
b can vary between 0..254 and I guess the author wants 0% only represented by `b=0`. `b=1` is a very low brightness but obviously not 0%, hence the manual 1%.